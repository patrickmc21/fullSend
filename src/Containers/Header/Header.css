.app-header {
  border-bottom: 1px solid #292727;
  height: 100px;
  background-color: #515c28;
}

.logo-link {
  outline: none;
}

.logo {
  float: left;
  margin: 20px 20px;
  width: 300px;
  background-image: url('./images/logo-img.png');
  background-size: 64px 60px;
  background-repeat: no-repeat;
  background-position: 210px 0px;
  color: #FFFFFE;
  font-family: 'Sedgwick Ave Display', cursive;
  font-size: 64px;
}

.user-info {
  float: right;
  border-radius: 25px;
  margin: 10px 10px;
  height: 80px;
  width: 250px;
  background-color: rgba(114,113,103, 0.4);
}

.user-profile-picture {
  float: right;
  margin: 10px 10px;
  width: 65px;
  border-radius: 30px;
}

.user-name,
.user-location {
  text-align: right;
  font-family: 'Raleway', sans-serif;
  color: #FFFFFE;
}

.user-name {
  margin: 25px 0 3px 0;
}