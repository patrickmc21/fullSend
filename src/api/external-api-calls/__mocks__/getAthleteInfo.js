import { mockUserInfo } from '../../../mock-data/mock-data';

const getAthleteInfo = jest.fn()
  .mockImplementation(() => {
    return Promise.resolve(
      mockUserInfo
    )
  });

export default getAthleteInfo;