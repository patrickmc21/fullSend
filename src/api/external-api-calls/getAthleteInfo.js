const getAthleteInfo = async (token) => {
  const rootUrl = 'https://www.strava.com/api/v3/athlete'
  const optionsObject = {
    method: 'GET',
    headers: {
      Authorization: `Bearer ${token}`,
      "Content-Type": 'application/json' 
    }
  };

  try {
    const response = await fetch(rootUrl, optionsObject);
    const data = await response.json();
    return data;
  } catch (error) {
    throw error.message;
  }
};

export default getAthleteInfo