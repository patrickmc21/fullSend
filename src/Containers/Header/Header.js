import React, { Component } from 'react';
import { connect } from 'react-redux';
import { dispatch } from 'redux';
import { NavLink, withRouter } from 'react-router-dom';
import PropTypes from 'prop-types';

import * as actions from '../../Actions';
import getAthleteInfo from  '../../api/external-api-calls/getAthleteInfo';

import './Header.css';

export class Header extends Component {
  constructor(props) {
    super(props)
    this.state = {
      name: '',
      location: '',
      img: ''
    }
  }

  componentDidUpdate = async () => {
    const { token } = this.props.user;
    if (!this.state.name && token) {
      const user = await getAthleteInfo(token);
      this.setState({
        name: `${user.firstname} ${user.lastname}`,
        location: `${user.city}, ${user.state}`,
        img: user.profile_medium
      })
    }
  }

  render() {

    const { name, location, img } = this.state;
    return(
      <header className='app-header'>
        <NavLink 
          to='/main'
          className='logo-link'
        >
          <h1 className='logo'>fullSend</h1>
        </NavLink>
        {name.length > 0 &&
        <aside className='user-info'>
          <img 
            src={img} 
            alt='user profile'
            className='user-profile-picture' />
          <h4 className='user-name'>{name}</h4>
          <h5 className='user-location'>{location}</h5>
        </aside>}
      </header>
    )
  }
};

Header.propTypes = {
  user: PropTypes.object,
  logoutUser: PropTypes.func
}

export const mapStateToProps = state => ({
  user: state.user
});

export const mapDispatchToProps = dispatch => ({
  logoutUser: id => dispatch(actions.logoutUser(id))
});

export default withRouter(connect(mapStateToProps, mapDispatchToProps)(Header));