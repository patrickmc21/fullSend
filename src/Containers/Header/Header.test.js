import React from 'react';
import { shallow } from 'enzyme';
import { Header, mapStateToProps, mapDispatchToProps } from './Header';
import getAthleteInfo from  '../../api/external-api-calls/getAthleteInfo';
jest.mock('../../api/external-api-calls/getAthleteInfo');

import { mockUserInfo, mockUser } from '../../mock-data/mock-data';

describe('Header', () => {

  let wrapper;
  let mockedUser;
  let mockedLogout;
  let mockedClearRides;
  let mockedChangeMonth;

  beforeEach(() => {
    mockedUser = mockUser;
    mockedLogout = jest.fn();
    mockedClearRides = jest.fn();
    mockedChangeMonth = jest.fn();
    wrapper = shallow(
      <Header 
        user={mockedUser} 
        logoutUser={mockedLogout}
        clearRides={mockedClearRides}
        changeMonth={mockedChangeMonth} />
    );
  });

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should match the snapshot after userInfo arrives', () => {
    wrapper.instance().componentDidUpdate();
    expect(wrapper).toMatchSnapshot();
  });

  it('should not update state if no user is present', () => {
    mockedUser = {name: '', token: '', id: null};
    wrapper = shallow(<Header user={mockedUser}/>);
    wrapper.instance().componentDidUpdate();
    expect(wrapper).toMatchSnapshot();
  });

  it('should call logoutUser on handleLogout', () => {
    const expected = mockUser.id;
    wrapper.instance().handleLogout();
    expect(mockedLogout).toHaveBeenCalledWith(expected);
  });

  it('should call clearRides on handleLogout', () => {
    const expected = mockUser.id;
    wrapper.instance().handleLogout();
    expect(mockedClearRides).toHaveBeenCalledWith(expected);
  });

  it('should call changeMonth on handleClick', () => {
    const expected = 'All';
    wrapper.instance().handleClick();
    expect(mockedChangeMonth).toHaveBeenCalledWith(expected);
  });
});

describe('mapStateToProps', () => {
  it('should return user state', () => {
    const mockState = {user: mockUser};
    const expected = mockUser;
    const mapped = mapStateToProps(mockState);
    expect(mapped.user).toEqual(expected);
  });
});

describe('mapDispatchToProps', () => {

  let mockDispatch;
  let mapped;

  beforeEach(() => {
    mockDispatch = jest.fn();
    mapped = mapDispatchToProps(mockDispatch);
  });

  it('should map logoutUser to props', () => {
    const expected = {
      type: 'LOGOUT_USER',
      id: mockUser.id
    };
    mapped.logoutUser(mockUser.id);
    expect(mockDispatch).toHaveBeenCalledWith(expected);
  });

  it('should map clearRides to props', () => {
    const expected = {
      type: 'CLEAR_RIDES',
      id: mockUser.id
    };
    mapped.clearRides(mockUser.id);
    expect(mockDispatch).toHaveBeenCalledWith(expected);
  });

  it('should map changeMonth to props', () => {
    const expected = {
      type: 'CHANGE_MONTH',
      month: 'May'
    };
    mapped.changeMonth('May');
    expect(mockDispatch).toHaveBeenCalledWith(expected);
  });
});