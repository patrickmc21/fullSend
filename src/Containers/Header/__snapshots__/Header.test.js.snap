// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header should match the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Header
    logoutUser={[Function]}
    user={
        Object {
            "id": 12,
            "name": "Pat",
            "token": "2y589wyhgfiouqehr8to8",
          }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <NavLink
          activeClassName="active"
          ariaCurrent="true"
          className="logo-link"
          to="/main"
>
          <h1
                    className="logo"
          >
                    fullSend
          </h1>
</NavLink>,
        false,
      ],
      "className": "app-header",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "activeClassName": "active",
          "ariaCurrent": "true",
          "children": <h1
            className="logo"
>
            fullSend
</h1>,
          "className": "logo-link",
          "to": "/main",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "fullSend",
            "className": "logo",
          },
          "ref": null,
          "rendered": "fullSend",
          "type": "h1",
        },
        "type": [Function],
      },
      false,
    ],
    "type": "header",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <NavLink
            activeClassName="active"
            ariaCurrent="true"
            className="logo-link"
            to="/main"
>
            <h1
                        className="logo"
            >
                        fullSend
            </h1>
</NavLink>,
          false,
        ],
        "className": "app-header",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "activeClassName": "active",
            "ariaCurrent": "true",
            "children": <h1
              className="logo"
>
              fullSend
</h1>,
            "className": "logo-link",
            "to": "/main",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "fullSend",
              "className": "logo",
            },
            "ref": null,
            "rendered": "fullSend",
            "type": "h1",
          },
          "type": [Function],
        },
        false,
      ],
      "type": "header",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`Header should match the snapshot after userInfo arrives 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Header
    logoutUser={[Function]}
    user={
        Object {
            "id": 12,
            "name": "Pat",
            "token": "2y589wyhgfiouqehr8to8",
          }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <NavLink
          activeClassName="active"
          ariaCurrent="true"
          className="logo-link"
          to="/main"
>
          <h1
                    className="logo"
          >
                    fullSend
          </h1>
</NavLink>,
        false,
      ],
      "className": "app-header",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "activeClassName": "active",
          "ariaCurrent": "true",
          "children": <h1
            className="logo"
>
            fullSend
</h1>,
          "className": "logo-link",
          "to": "/main",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "fullSend",
            "className": "logo",
          },
          "ref": null,
          "rendered": "fullSend",
          "type": "h1",
        },
        "type": [Function],
      },
      false,
    ],
    "type": "header",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <NavLink
            activeClassName="active"
            ariaCurrent="true"
            className="logo-link"
            to="/main"
>
            <h1
                        className="logo"
            >
                        fullSend
            </h1>
</NavLink>,
          false,
        ],
        "className": "app-header",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "activeClassName": "active",
            "ariaCurrent": "true",
            "children": <h1
              className="logo"
>
              fullSend
</h1>,
            "className": "logo-link",
            "to": "/main",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "fullSend",
              "className": "logo",
            },
            "ref": null,
            "rendered": "fullSend",
            "type": "h1",
          },
          "type": [Function],
        },
        false,
      ],
      "type": "header",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`Header should not update state if no user is present 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Header
    user={
        Object {
            "id": null,
            "name": "",
            "token": "",
          }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <NavLink
          activeClassName="active"
          ariaCurrent="true"
          className="logo-link"
          to="/main"
>
          <h1
                    className="logo"
          >
                    fullSend
          </h1>
</NavLink>,
        false,
      ],
      "className": "app-header",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "activeClassName": "active",
          "ariaCurrent": "true",
          "children": <h1
            className="logo"
>
            fullSend
</h1>,
          "className": "logo-link",
          "to": "/main",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "fullSend",
            "className": "logo",
          },
          "ref": null,
          "rendered": "fullSend",
          "type": "h1",
        },
        "type": [Function],
      },
      false,
    ],
    "type": "header",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <NavLink
            activeClassName="active"
            ariaCurrent="true"
            className="logo-link"
            to="/main"
>
            <h1
                        className="logo"
            >
                        fullSend
            </h1>
</NavLink>,
          false,
        ],
        "className": "app-header",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "activeClassName": "active",
            "ariaCurrent": "true",
            "children": <h1
              className="logo"
>
              fullSend
</h1>,
            "className": "logo-link",
            "to": "/main",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "fullSend",
              "className": "logo",
            },
            "ref": null,
            "rendered": "fullSend",
            "type": "h1",
          },
          "type": [Function],
        },
        false,
      ],
      "type": "header",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
